tools/gce-google-keys-to-cmek/main.py

```diff
@@ -0,0 +1,207 @@
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import googleapiclient
+import googleapiclient.discovery
+import time
+import re
+import sys
+
+def main():
+    parser = argparse.ArgumentParser(description='Convert disks attached to a GCE instance from Google-managed encryption keys to customer-managed encryption keys.')
+    parser.add_argument('--project', required=True, dest='project', action='store', type=str, help='Project containing the GCE instance.')
+    parser.add_argument('--zone', required=True, dest='zone', action='store', type=str, help='Zone containing the GCE instance.')
+    parser.add_argument('--instance', required=True, dest='instance', action='store', type=str, help='Instance name.')
+    parser.add_argument('--key-ring', required=True, dest='keyRing', action='store', type=str, help='Name of the key ring containing the key to encrypt the disks. Must be in the same zone as the instance.')
+    parser.add_argument('--key-name', required=True, dest='keyName', action='store', type=str, help='Name of the key to encrypt the disks. Must be in the same zone as the instance.')
+    parser.add_argument('--key-version', required=True, dest='keyVersion', action='store', type=int, help='Version of the key to encrypt the disks.')
+    parser.add_argument('--destructive', dest='destructive', action='store_const', const=True, default=False, help='Upon completion, delete source disks and snapshots created during migration process.')
+    args = parser.parse_args()
+
+    migrate_instance_to_cmek(args.project, args.zone, args.instance, args.keyRing, args.keyName, args.keyVersion, args.destructive)
+
+def migrate_instance_to_cmek(project, zone, instance, keyRing, keyName, keyVersion, destructive):
+    start = time.time()
+
+    zone_regexp = r'^(\w\w-\w*\d)-(\w)$'
+    region = re.search(zone_regexp, zone).group(1)
+
+    compute = googleapiclient.discovery.build('compute', 'v1')
+
+    # TODO: Consider if we can use compute.disks().list() and do disk checking
+    # before we decide to stop the VM instance.
+    stop_instance(compute, project, zone, instance)
+    disks = get_instance_disks(compute, project, zone, instance)
+    for sourceDisk in disks:
+        disk_regexp = r'^https:\/\/www\.googleapis\.com\/compute\/v1\/projects\/(.*?)\/zones\/(.*?)\/disks\/(.*?)$'
+        disk_url = sourceDisk['source']
+        existing_disk_name = re.search(disk_regexp,disk_url).group(3)
+
+        if 'diskEncryptionKey' in sourceDisk:
+            print('Skipping {0}, already encrypyed with {1}', existing_disk_name, sourceDisk['diskEncryptionKey'])
+            continue
+
+        snapshot_name = existing_disk_name + '-goog-to-cmek'
```

Comment:
nit: let's use a format string.
```suggestion
        snapshot_name =  '{}-goog-to-cmek'.format(existing_disk_name)
```